src/Components/Facility/FacilityCNS.tsx

```diff
@@ -237,7 +274,11 @@ export default function FacilityCNS({ facilityId }: { facilityId: string }) {
       )}
       <div className="mt-4 grid grid-cols-1 lg:grid-cols-3 gap-1">
         {monitors
-          ?.filter((m) => m.asset.location_object.id === location?.id)
+          ?.filter((m) =>
+            defaultShowAllLocation
+              ? true
+              : m.asset.location_object.id === location?.id
+          )
```

Comment:
Return monitors directly without `.filter` if `defaultShowAllLocation` so that it won't needlessly iterate.